README.md

```diff
@@ -94,6 +94,23 @@ If used, one may want to use the first value of this setting to configure
 daffodilVersion := daffodilPackageBinVersions.value.head
 ```
 
+## Layers and User Defined Functions
+
+If your schema project builds a Daffodil layer or user defined function, then
+set the `daffodilBuildsLayer` or `daffodilBuildsUDF` setting to true,
+respectively. For example:
+
+```scala
+daffodilBuildsLayer := true
+
+daffodilBuildsUDF := true
+```
+
+Setting either of these values to true adds additional dependencies needed to
+build the component. Note that this also sets the SBT `crossPaths` setting to
+`true`, which causes the scala version to be included in that jar file name
+since layer and UDF jars are specific to the scala version used to build them.
```

Comment:
If the UDF is written in Java then no crossPaths would be needed right?

A goal for the revised layer API should be that one can write a layer in Java I suppose. 

Do we need to know if Java or Scala are being used?

Reply:
Yeah, I think you're right.

Maybe instead of using these settings we should just check if `Compile / sources` contains any files that end in *.scala. If any exist then we set `crossPaths` to true. This has the advantage that if a schema project has any scala source files at all, even if they aren't using UDFs or layers, that they get the correct _2.12 since they do use scala.

Reply:
Or add a scalaCrossPaths option which if false shuts off the cross paths, but allows the other settings changes?

Perhaps it's just allowScala which if not present or false scala will not be allowed in the project?

Reply:
I'm fine with just adding one or two more sentences to the README explaining that if turning on crossPaths isn't actually desired or needed (say because the UDF or layer is written in Java, not Scala), simply set `crossPaths := false` in build.sbt.

Reply:
> Or add a scalaCrossPaths option which if false shuts off the cross paths, but allows the other settings changes?
>
> Perhaps it's just allowScala which if not present or false scala will not be allowed in the project?

These feel like it might start complicating things too much for users. The goal is to keep things as simple as possible for schema authors and giving them reasonable defaults. I think the new `daffodilBuildsUDF` and `daffodilBuildsLayer` are useful since they provide extra dependencies so users don't have to think about it, but beyond that I think starts making things complicated and causes users to have to remember more and more settings. We're already starting to build up a lot of settings.

If a project wants to disallow scala files, then maybe the suggested way is to scan for it in code reviews or by adding *.scala to .gitignore. And if they don't like our crossPaths defaults, they can manually override it with the existing SBT setting.

I like @tuxji's idea of just documnting that if a project has layers/udfs that don't use any Scala and they don't want the _2.12 then to set crossPaths to false in build.sbt.